### PR TITLE
improved test coverage and updated nusje2000/dependency-graph

### PR DESCRIPTION
**Pull request by nusje2000**

### CODE REVIEW

File: tests/Validator/ValidatorTest.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        $rule = $this->createMock(RuleInterface::class);
+        $rule->expects(self::once())->method('execute')->willReturn(
+            new ViolationCollection([
+                $this->createStub(ViolationInterface::class),
+            ])
+        );
+
+        $validator = new Validator(new RuleCollection([$rule]));
+        $output = $validator->validate(new DependencyGraph('/', new PackageCollection()));
```
Comment (dbrekelmans) [inline tests/Validator/ValidatorTest.php]: Maybe not within the scope of this PR, but I would not necessarily expect `validate()` to return anything. Maybe you could consider renaming the function to more explicitly communicate the intent (it's not just validating but also returning x)

For inspiration you could take a look at the symfony validator component. I think the similar function is simply called `getErrors`

File: src/Validator/ViolationCollection.php [hunk 2]
```diff
@@ -27,6 +27,15 @@ public function getMessages(): StringCollection
 
         return new StringCollection($messages);
     }
+    public function getFormattedMessages(): StringCollection
+    {
+        /** @var array<string> $messages */
+        $messages = $this->map(static function (ViolationInterface $violation) {
```
Comment (nvdbeek) [inline src/Validator/ViolationCollection.php]: PHPStan dont expect a return type for this closure?

File: tests/Validator/ValidatorTest.php (hunk 1 above)
Comment (nusje2000) [reply tests/Validator/ValidatorTest.php]: Created #10

File: src/Validator/ViolationCollection.php (hunk 2 above)
Comment (nusje2000) [reply src/Validator/ViolationCollection.php]: Somehow it does not, but I don't know why. I'll add it.

**Pull request merged**